### DISCUSSION TITLE
Delete files after deleting file field

### DISCUSSION THREAD

**Discussion by diogomartino** [pocketbase/pocketbase#3517] [Q&A]
In a situation where I have a data collection containing a file field with existing data associated with files, deleting the file field should logically result in the deletion of all files associated with that field. However, I made a backup before and after deleting the field, and I noticed that the backup file size remained unchanged. This suggests that the files may still be present for some unknown reason. Is there a way to remove them? `v0.18.7`

**Comment by ganigeorgiev** [comment]
This is a known issue - https://github.com/pocketbase/pocketbase/issues/305#issuecomment-1213439223.

Unfortunately there is no easy way to remove the files on collection `file` field removal at the moment.

The files would be deleted if you delete the collection or its records.

If your collection doesn't have any other `file` fileds, one hacky workaround could be to create a duplicate of your collection, move the records from the old collection to the new one with the sqlite3 cli tool like `sqlite3 ./pb_data/data.db 'INSERT INTO duplicate SELECT * FROM old'` and then delete the old collection.

**Reply by diogomartino** [reply]
So, deleting the collection would delete the old files even if the file field was already deleted before?

**Reply by ganigeorgiev** [reply]
Yes, deleting the collection (or its records) should delete the associated files.

Internally the files are stored with the following dir/prefix structure: `/:collectionId/:recordId/:filename`.
When deleting a collection or a record we delete directly the `/:collectionId` or `/:collectionId/:recordId` directory/prefix.

**Reply by diogomartino** [reply]
Great, that should be good enough for me. Thanks!
